### PR TITLE
Add offset pushdown to parquet

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

A follow up to https://github.com/apache/arrow-rs/pull/3633 that also allows pushing down an offset

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by thinkharderdev — APPROVED** [review]
LGTM

File: parquet/src/arrow/arrow_reader/selection.rs [hunk 1]
```diff
@@ -372,29 +371,63 @@ impl RowSelection {
...
+            Some(idx) => idx,
+            None => {
+                self.selectors.clear();
+                return self;
+            }
+        };
+
+        let mut selectors = Vec::with_capacity(self.selectors.len() - split_idx + 1);
+        selectors.push(RowSelector::skip(skipped_count + offset));
+        selectors.push(RowSelector::select(selected_count - offset));
```
Comment (viirya) [inline parquet/src/arrow/arrow_reader/selection.rs:408]: `skip` and `select` cannot be interleaved? i.e. all `skip` selections are in front of `select` selection?

Comment (tustvold) [reply parquet/src/arrow/arrow_reader/selection.rs:408]: This operator is skipping the first offset rows, so will always start with a skip, the remaining selection afterwards can be interleaved

Comment (viirya) [reply parquet/src/arrow/arrow_reader/selection.rs:408]: Hmm, I meant the way you calculate skpped_count and selected_count. If original selectors are mixed with `skip` and `select`, is the logic still correct?

E.g., 1st selector `skip` 5 row, 2nd selector `select` 1 row, 3rd selector `skip` 2 rows, 4 selector `select` 5 rows.

With `offset` 2, this produces 1st selector `skip` 7 + 2 rows, 2nd selector `select` 6 - 2 rows. But original 2nd selector's 1 row is skipped now.

Comment (viirya) [reply parquet/src/arrow/arrow_reader/selection.rs:408]: Oh, it's correct as skipped rows only count until selected count is larger than offset.

**Review by viirya — APPROVED** [review]

**Pull request merged**